### PR TITLE
backoff.TestRetry: increase initial backoff duration from 1ns to 50ms

### PR DESCRIPTION
**Pull request by RJPercival**
Avoids a race between waiting for the backoff duration and checking whether the context is done, which was making the test flaky.

### CODE REVIEW

File: client/backoff/backoff_test.go [hunk 1]
```diff
@@ -92,8 +92,8 @@ func TestJitter(t *testing.T) {
 
 func TestRetry(t *testing.T) {
 	b := Backoff{
-		Min:    time.Duration(1),
-		Max:    time.Duration(100),
+		Min:    time.Duration(50) * time.Millisecond,
+		Max:    time.Duration(200) * time.Millisecond,
```
Comment (daviddrysdale) [inline client/backoff/backoff_test.go]: These can just be like `Min: 50 * time.Millisecond` can't they? 

(Looks like there's lots of unnecessary casts in this file)

Comment (RJPercival) [inline client/backoff/backoff_test.go]: Done.

**Comment by gdbelvin** [comment]
If the test is flaky, does that mean that the code itself has a race condition?  
Seems like having a 1 ns delay might actually be a good thing as a test?

**Comment by RJPercival** [comment]
David (in an offline conversation) wasn't a fan of having code that protects against the race condition when it'll only exhibit itself when the backoff duration is incredibly short, which was highly unlikely to be the case in a real scenario.

**Comment by daviddrysdale** [comment]
AIUI, the code doesn't have a race condition; if you cancel at around the same time that a retry timer pops, it's a roll of the dice as to whether you get a retry or not -- but there's no raciness, just uncertainty.  So the tests have been changed to a more certain scenario.

**Pull request merged**